addons/mail/static/src/models/thread/thread.js

```diff
@@ -452,18 +459,32 @@ function factory(dependencies) {
 
         /**
          * Mark the specified conversation as read/seen.
+         * @param {mail.message} [message] the message to be considered as last seen, by default
+         * last message of the thread.
          */
-        async markAsSeen() {
-            if (this.message_unread_counter === 0) {
+        markAsSeen(message) {
+            if (!message) {
+                message = this.lastMessage;
+            }
+            if (!message) {
                 return;
             }
+            if (this.pendingSeenMessageId && message.id <= this.pendingSeenMessageId) {
+                return;
+            }
+            if (this.seen_message_id && message.id <= this.seen_message_id) {
+                return;
+            }
+            this.update({ pendingSeenMessageId: message.id });
             if (this.model === 'mail.channel') {
-                const seen_message_id = await this.async(() => this.env.services.rpc({
+                this.env.services.rpc({
```

Comment:
If the result of the RPC is ignored, the python method should not return a value in the first place.

I would still expect an `await` for consistency.